tools/codegen/gen.py

```diff
@@ -273,7 +273,7 @@ def func(f: NativeFunction) -> Optional[str]:
             assert returns_type == dispatcher.returns_type(f.func.returns)
             dispatcher_args = dispatcher.arguments(f.func)
             dispatcher_args_types_str = ', '.join(map(lambda a: a.type, dispatcher_args))
-            if dispatch is None or dispatch == 'Math':
+            if dispatch is None or dispatch == 'Math' or dispatch == 'DefaultBackend':
```

Comment:
Given that this grouping (Math and DefaultBackend) appears twice, once here and once in the assert in `model.py`, there might be a benefit to making a global in model.py like `TYPE_DEFAULT_DISPATCH_KEYS` (or a better name) and use it in both places - that'll make it visible to the data model and help to avoid (or at least highlight) ad-hoc changes to this logic.

Reply:
Ah although they show up in both files, the semantics are not necessarily connected? I feel the throw semantics in model.py is mainly an additional rule we add due to the precedence between Math and DefaultBackend and that's not strongly tied to Typedefault codegen.

Reply:
TBH the `TypeDefault` naming here is kind of a vestige and should get refactored away at some point.